electrum/wallet.py

```diff
@@ -2841,9 +2842,11 @@ def create_request(self, amount_sat: Optional[int], message: Optional[str], exp_
         address = address or None  # converts "" to None
         exp_delay = exp_delay or 0
         timestamp = int(Request._get_cur_time())
-        payment_hash = None  # type: Optional[bytes]
-        if self.has_lightning():
+        if address is None:
+            assert self.has_lightning()
             payment_hash = self.lnworker.create_payment_info(amount_msat=amount_msat, write_to_disk=False)
+        else:
+            payment_hash = None
```

Comment:
This makes the `WALLET_BOLT11_FALLBACK` config setting useless. That setting is exposed to the GUIs.

From the context, and also seeing as you removed the unit test that was checking it such addresses are reused as expected, I assume you just forgot to remove this config key?

Reply:
right, we should remove that config key